### PR TITLE
JAMES-2513 Disable DomainList localhost translation when autoDetect is false

### PR DESCRIPTION
**Pull request by chibenwa**
This may be the root cause of some kubernetes DNS + DomainList fails

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[c793b9ad6f299161244238625dc6c4c40d2db1f4]   XMLDomainListTest.configureShouldNotFailWhenConfiguringDefaultDomain:162 
[c793b9ad6f299161244238625dc6c4c40d2db1f4] Expected size:<3> but was:<2> in:
[c793b9ad6f299161244238625dc6c4c40d2db1f4] <[Domain : localhost, Domain : domain1]>
```

**Comment by chibenwa** [comment]
```
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] Expecting:
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <[]>
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] to contain exactly (and in same order):
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <["name1", "name2"]>
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] but could not find the following elements:
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <["name1", "name2"]>
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] 	at org.apache.james.queue.api.ManageableMailQueueContract.concurrentClearShouldNotAlterBrowsingWhenDequeue(ManageableMailQueueContract.java:282)
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] Results :
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] Failed tests: 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   ManageableMailQueueContract.concurrentClearShouldNotAlterBrowsingWhenDequeue:282 
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] Expecting:
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <[]>
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] to contain exactly (and in same order):
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <["name1", "name2"]>
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307] but could not find the following elements:
[0ae3f3164bd81bd094485ddfef4ee7a9342d2307]   <["name1", "name2"]>
```

test this please

**Comment by trantienduchn** [comment]
```
[2af816ea014f3b369a0cf099e9a968eb53d04449] Failed tests: 
[2af816ea014f3b369a0cf099e9a968eb53d04449]   AbstractJamesServerTest.hostnameShouldBeRetrievedWhenRestarting:76 expected:<"[0e9cc6aa41d6]"> but was:<"[localhost]">
[2af816ea014f3b369a0cf099e9a968eb53d04449]   AbstractJamesServerTest.hostnameShouldBeUsedAsDefaultDomain:67 expected:<"[0e9cc6aa41d6]"> but was:<"[localhost]">
[2af816ea014f3b369a0cf099e9a968eb53d04449]   AbstractJamesServerTest.hostnameShouldBeRetrievedWhenRestarting:76 expected:<"[0e9cc6aa41d6]"> but was:<"[localhost]">
[2af816ea014f3b369a0cf099e9a968eb53d04449]   AbstractJamesServerTest.hostnameShouldBeUsedAsDefaultDomain:67 expected:<"[0e9cc6aa41d6]"> but was:<"[localhost]">
[2af816ea014f3b369a0cf099e9a968eb53d04449] 
[2af816ea014f3b369a0cf099e9a968eb53d04449] Tests run: 17, Failures: 4, Errors: 0, Skipped: 0
```

File: server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java [hunk 1]
```diff
@@ -87,6 +87,10 @@ public void configure(DomainListConfiguration domainListConfiguration) throws Co
         addConfiguredDomains(domainListConfiguration.getConfiguredDomains());
     }
 
+    public void configure(DomainListConfiguration.Builder configurationBuilder) throws ConfigurationException {
```
Comment (mbaechler) [inline server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java]: belongs to previous commit, right ?

Comment (chibenwa) [reply server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java]: When you do such comment, can you include current commit SHA please? It makes finding the incriminated commit easier.

Comment (chibenwa) [reply server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java]: I reworked the history.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**